tests/test_loaders.py

```diff
@@ -41,3 +53,122 @@ def test_load_sound_speed(all_files_dict):
     assert isinstance(svdf, DataFrame)
     assert {SP_DEPTH, SP_SOUND_SPEED} == set(svdf.columns.values.tolist())
     assert svdf.dtypes[SP_DEPTH] == float64 and svdf.dtypes[SP_SOUND_SPEED] == float64
+
+
+@pytest.fixture
+def transponder_ids() -> List[str]:
+    config = load_configuration(TEST_DATA_FOLDER / "config.yaml")
+    transponders = config.solver.transponders
+    return [t.pxp_id for t in transponders]
+
+
+@pytest.mark.parametrize(
+    "is_j2k, time_scale",
+    [(True, "tt"), (False, "tt")],
+)
+def test_load_j2k_travel_times(
+    transponder_ids, all_files_dict_j2k_travel_times, is_j2k, time_scale
+):
+    if is_j2k:
+        loaded_travel_times = load_travel_times(
+            files=all_files_dict_j2k_travel_times["travel_times"],
+            transponder_ids=transponder_ids,
+            is_j2k=is_j2k,
+            time_scale=time_scale,
+        )
+
+        # raw_travel_times contains the expected df
+        raw_travel_times = concat(
+            [
+                read_csv(i, delim_whitespace=True, header=None)
+                for i in all_files_dict_j2k_travel_times["travel_times"]
+                if PARSED_FILE not in i
+            ]
+        ).reset_index(drop=True)
+
+        expected_columns = [TT_TIME, *transponder_ids]
+        column_num_diff = len(expected_columns) - len(raw_travel_times.columns)
+        if column_num_diff < 0:
+            raw_travel_times = raw_travel_times.iloc[:, :column_num_diff]
+        raw_travel_times.columns = expected_columns
+
+        # Assert that df returned from "loaded_travel_times()" matches parameters of expected df
+        assert isinstance(loaded_travel_times, DataFrame)
+        assert all(
+            is_float_dtype(loaded_travel_times[column])
+            for column in [*transponder_ids, TT_TIME]
+        )
+        assert loaded_travel_times.shape == raw_travel_times.shape
+        assert set(loaded_travel_times.columns.values.tolist()) == set(
+            raw_travel_times.columns.values.tolist()
+        )
+
+        # Verify microsecond to second conversion for each transponder_id column
+        assert loaded_travel_times[transponder_ids].equals(
+            raw_travel_times[transponder_ids].apply(lambda x: x * 1e-6)
+        )
+    else:
+        # load_travel_times() should raise Exception
+        # if called with is_j2k=False on j2k type travel time files
+        with pytest.raises(AttributeError):
+            _ = load_travel_times(
+                files=all_files_dict_j2k_travel_times["travel_times"],
+                transponder_ids=transponder_ids,
+                is_j2k=is_j2k,
+                time_scale=time_scale,
+            )

```

Comment:
This `else` is very hidden and it's repeated down below. I think it would be more clear if this else statement should go up top by doing like:

```python
if not is_j2k:
    # load_travel_times() should raise Exception
    # if called with is_j2k=False on j2k type travel time files
    with pytest.raises(AttributeError):
        _ = load_travel_times(
            files=all_files_dict_j2k_travel_times["travel_times"],
            transponder_ids=transponder_ids,
            is_j2k=is_j2k,
            time_scale=time_scale,
        )
    return

# The rest of code goes below
loaded_travel_times = load_travel_times(
    files=all_files_dict_j2k_travel_times["travel_times"],
    transponder_ids=transponder_ids,
    is_j2k=is_j2k,
    time_scale=time_scale,
)
```

Reply:
Done!